community/CONTRIBUTING.md

```diff
@@ -5,6 +5,195 @@ Thanks for considering contributing to OPEA project. The contribution process is
+
+
+```
+GenAIComps
+├── comps
+│   ├── agent
+│   ├── asr
+│   ├── chathistory
+│   ├── cores
+│   │   ├── mega     #orchestrator, gateway, micro_service class code
+│   │   ├── proto    #api protocol
+│   │   └── telemetry
+│   ├── dataprep
+│   ├── embeddings
+│   ├── feedback_management
+│   ├── finetuning
+│   ├── guardrails
+│   ├── intent_detection
+│   ├── knowledgegraphs
+│   ├── llms
+│   ├── lvms
+│   ├── nginx
+│   ├── prompt_registry
+│   ├── ragas
+│   ├── reranks
+│   ├── retrievers
+│   ├── tts
+│   ├── vectorstores
+│   └── web_retrievers
+└── tests
+    ├── agent
+    ├── asr
+    ├── chathistory
+    ├── cores
+    ├── dataprep
+    ├── embeddings
+    ├── feedback_management
+    ├── finetuning
+    ├── guardrails
+    ├── intent_detection
+    ├── llms
+    ├── lvms
+    ├── nginx
+    ├── prompt_registry
+    ├── reranks
+    ├── retrievers
+    ├── test_finetuning_llm_pretraining.sh
+    ├── test_finetuning_rerank.sh
+    ├── test_reranks_mosec-neuralspeed.sh
+    ├── tts
+    ├── vectorstores
+    └── web_retrievers
+```
+
+2. Follow the folder structure in the TEI embedding component below:
+
+```
+GenAIComps
+├── comps
+│   └── embeddings
+│       ├── __init__.py
+│       └── tei     #vendor name or serving framework name
+│           ├── langchain
+│           │   ├── docker_compose_embedding.yaml  #could be multiple for different HW
+│           │   ├── Dockerfile        
+│           │   ├── Dockerfile_AMD_GPU         
+│           │   ├── Dockerfile_Nvidia_GPU      
+│           │   ├── embedding_tei.py    # definition and registration of microservice
+│           │   ├── README.md
+│           │   └── requirements.txt
+│           └── llama_index
+│               └── . . .
+├── tests
+│   └── embeddings
+│       ├── test_embeddings_tei_langchain.sh
+│       ├── test_embeddings_tei_langchain_amd_gpu.sh     
+│       └── test_embeddings_tei_llama_index.sh
+└── README.md
+
+```
+- **File Descriptions**:
+  - `embedding_tei.py`: This file defines and registers the microservice. It serves as the entrypoint of the Docker container. Refer to [whisper ASR](https://github.com/opea-project/GenAIComps/tree/main/comps/asr/whisper) for a simple example or [TGI](https://github.com/opea-project/GenAIComps/blob/main/comps/llms/text-generation/tgi/llm.py) for a more complex example that required adapting to OpenAI API.
+  - `requirements.txt`: This file is used by Docker to install the necessary dependencies.
+  - `Dockerfile` and optionally `docker_compose_yaml`: These files are used to generate the service container image. Please follow naming conventions:
+    - Dockerfile: `Dockerfile_[vendor]_[hardware]`
```

Comment:
`Dockerfile.[vendor]_[hardware]` vendor and hardware should be in lower-case.

Reply:
done